classes/oai/PKPOAIDAO.inc.php

```diff
@@ -291,10 +273,12 @@ public function _doCommonOAIFromRowOperations($record, $row)
      * @param $set string
      * @param $submissionId int optional
      * @param $orderBy string UNFILTERED
+     * @param $offset int? Query offset
+     * @param $limit int? Query limit
      *
-     * @return Iterable
+     * @return \Illuminate\Database\Query\Builder
      */
-    abstract public function _getRecordsRecordSet($setIds, $from, $until, $set, $submissionId = null, $orderBy = 'journal_id, submission_id');
+    abstract public function _getRecordsRecordSetQuery($setIds, $from, $until, $set, $submissionId = null, $orderBy = 'journal_id, submission_id', $offset = null, $limit = null);
 }
```

Comment:
I think we do not need `$offset` and `$limit` parameters for this function -- as you did above the total/count, offset and limit are called on the results of the function?